src/ReverseDiffPrototype.jl

```diff
@@ -19,12 +19,30 @@ type TapeReal{tag,T<:Real} <: Real
     val::T
     adj::T
 end
+typealias TapeValue{tag, T} Union{TapeReal{tag, T}, Array{TapeReal{tag, T}}}
 
 TapeReal{tag<:Tag,T}(::Type{tag}, val::T) = TapeReal{tag,T}(val)
 
 @inline value(n::TapeReal) = n.val
 @inline adjoint(n::TapeReal) = n.adj
+
+# we can't use map here because we'll overload it below to record all ops
+@inline function value{tag,T<:Real}(A::Array{TapeReal{tag, T}})
+    out = similar(A, T)
+    for i in eachindex(A)
+        out[i] = A[i].val
+    end
+    out
+end
+@inline function adjoint{tag,T<:Real}(A::Array{TapeReal{tag, T}})
+    out = similar(A, T)
+    for i in eachindex(A)
+        out[i] = A[i].adj
+    end
+    out
+end
```

Comment:
Oh yeah. That makes sense.
On Sun, Jun 26, 2016 at 08:54 Jarrett Revels <notifications@github.com>
wrote:

> In src/ReverseDiffPrototype.jl
> <https://github.com/jrevels/ReverseDiffPrototype.jl/pull/6#discussion_r68502664>
> :
>
> > +
> > +# we can't use map here because we'll overload it below to record all ops
> > +@inline function value{tag,T<:Real}(A::Array{TapeReal{tag, T}})
> > +    out = similar(A, T)
> > +    for i in eachindex(A)
> > +        out[i] = A[i].val
> > +    end
> > +    out
> > +end
> > +@inline function adjoint{tag,T<:Real}(A::Array{TapeReal{tag, T}})
> > +    out = similar(A, T)
> > +    for i in eachindex(A)
> > +        out[i] = A[i].adj
> > +    end
> > +    out
> > +end
>
> Can we do A[i].val --> value(A[i]) and A[i].adj --> adjoint(A[i])?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jrevels/ReverseDiffPrototype.jl/pull/6/files/70b10963a614d625be466354ede2f5a2a4d934c7#r68502664>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADBM5-Shrnu64Vn4Gctz3CUmVLzx_0CEks5qPnaRgaJpZM4I-ZlG>
> .
>